common/lru_policy.go

```diff
@@ -0,0 +1,113 @@
+package common
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+)
+
+type CacheBlock struct {
+	sync.RWMutex
+	StartIndex int64
+	EndIndex   int64
+	Data       []byte
+	Last       bool //last block in the file?
+}
+
+//KeyPair: the list node containing both block key and cache block values
+type KeyPair struct {
+	key   int64
+	value *CacheBlock
+}
+
+//LRUCache definition for Least Recently Used Cache implementation
+type LRUCache struct {
+	sync.RWMutex
+	Capacity int64
+	List     *list.List              //DoublyLinkedList: node1->node2.... node:=KeyPair
+	Elements map[int64]*list.Element //blockKey:KeyPair
+	Occupied int64
```

Comment:
What is this variable? To keep track of the number of blocks in the cache?

Reply:
to keep track of the occupied space/capacity